packages/quick_actions/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java

```diff
@@ -60,6 +69,15 @@ public void onDetachedFromActivityForConfigChanges() {
     onDetachedFromActivity();
   }
 
+  @Override
+  public boolean onNewIntent(Intent intent) {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N_MR1) return false;
+    if (intent.hasExtra(MethodCallHandlerImpl.EXTRA_ACTION) && channel != null) {
+      channel.invokeMethod("launch", intent.getStringExtra(MethodCallHandlerImpl.EXTRA_ACTION));
+    }
+    return false;
+  }
+
   private void setupChannel(BinaryMessenger messenger, Context context, Activity activity) {
     channel = new MethodChannel(messenger, CHANNEL_ID);
     handler = new MethodCallHandlerImpl(context, activity);
```

Comment:
MethodCallHandlerImpl could take an object that implements an interface. Then, QuickActionsPlugin could implement the interface.  Lastly, pass the reference when it constructs `MethodCallHandlerImpl`.

Reply:
The interface implementation sets `isInitialized = true;`

Reply:
Thanks for the suggestion. But as per stuartmorgan@'s comment above, removed the field as it's not necessary.